crates/core_arch/src/x86/macros.rs

```diff
@@ -247,6 +247,491 @@ macro_rules! constify_imm4_sae {
     };
 }
 
+// Two mantissas parameters.
+// This macro enforces that.
+#[allow(unused)]
+macro_rules! constify_imm4_mantissas {
+    ($imm4:expr, $imm2:expr, $expand:ident) => {
+        #[allow(overflowing_literals)]
+        match ($imm4, $imm2) {
+            (0, 0) => $expand!(0, 0),
+            (0, 1) => $expand!(0, 1),
+            (0, 2) => $expand!(0, 2),
+            (0, 3) => $expand!(0, 3),
+            (1, 0) => $expand!(1, 0),
+            (1, 1) => $expand!(1, 1),
+            (1, 2) => $expand!(1, 2),
+            (1, 3) => $expand!(1, 3),
+            (2, 0) => $expand!(2, 0),
+            (2, 1) => $expand!(2, 1),
+            (2, 2) => $expand!(2, 2),
+            (2, 3) => $expand!(2, 3),
+            (3, 0) => $expand!(3, 0),
+            (3, 1) => $expand!(3, 1),
+            (3, 2) => $expand!(3, 2),
+            (3, 3) => $expand!(3, 3),
+            (4, 0) => $expand!(4, 0),
+            (4, 1) => $expand!(4, 1),
+            (4, 2) => $expand!(4, 2),
+            (4, 3) => $expand!(4, 3),
+            (5, 0) => $expand!(5, 0),
+            (5, 1) => $expand!(5, 1),
+            (5, 2) => $expand!(5, 2),
+            (5, 3) => $expand!(5, 3),
+            (6, 0) => $expand!(6, 0),
+            (6, 1) => $expand!(6, 1),
+            (6, 2) => $expand!(6, 2),
+            (6, 3) => $expand!(6, 3),
+            (7, 0) => $expand!(7, 0),
+            (7, 1) => $expand!(7, 1),
+            (7, 2) => $expand!(7, 2),
+            (7, 3) => $expand!(7, 3),
+            (8, 0) => $expand!(8, 0),
+            (8, 1) => $expand!(8, 1),
+            (8, 2) => $expand!(8, 2),
+            (8, 3) => $expand!(8, 3),
+            (9, 0) => $expand!(9, 0),
+            (9, 1) => $expand!(9, 1),
+            (9, 2) => $expand!(9, 2),
+            (9, 3) => $expand!(9, 3),
+            (10, 0) => $expand!(10, 0),
+            (10, 1) => $expand!(10, 1),
+            (10, 2) => $expand!(10, 2),
+            (10, 3) => $expand!(10, 3),
+            (11, 0) => $expand!(11, 0),
+            (11, 1) => $expand!(11, 1),
+            (11, 2) => $expand!(11, 2),
+            (11, 3) => $expand!(11, 3),
+            (12, 0) => $expand!(12, 0),
+            (12, 1) => $expand!(12, 1),
+            (12, 2) => $expand!(12, 2),
+            (12, 3) => $expand!(12, 3),
+            (13, 0) => $expand!(13, 0),
+            (13, 1) => $expand!(13, 1),
+            (13, 2) => $expand!(13, 2),
+            (13, 3) => $expand!(13, 3),
+            (14, 0) => $expand!(14, 0),
+            (14, 1) => $expand!(14, 1),
+            (14, 2) => $expand!(14, 2),
+            (14, 3) => $expand!(14, 3),
+            (15, 0) => $expand!(15, 0),
+            (15, 1) => $expand!(15, 1),
+            (15, 2) => $expand!(15, 2),
+            (15, 3) => $expand!(15, 3),
+            (_, _) => panic!("Invalid sae value"),
```

Comment:
Should this be invalid mantissa instead?

Reply:
When in doubt you should check what GCC accepts. It is generally closer to the spec than Clang.

Reply:
Just use (0..15)(0..3) like the C compilers.

Reply:
Yes